### PR TITLE
Support for Keycloak - Open Source Identity and Access Management

### PR DESCRIPTION
**Pull request by gtonic**
Support the (quarkus based) version of Keycloak IAM.

### CODE REVIEW

**Comment by adrianliechti** [comment]
great idea and mighty tool (although I am an ory hydra fan :) )

maybe some ideas for further improvements:
- set an initial password to a random one (just for the sake of training good habits to users)
see here: https://github.com/adrianliechti/devkit/blob/main/pkg/catalog/mariadb/mariadb.go#L43

- hint the username / password + url to help developers how to connect (info call) -> extract env variable with password; build an url
see here: https://github.com/adrianliechti/devkit/blob/main/pkg/catalog/minio/minio.go#L73

- since the documentation is that great (haha), the idea of the console endpoint was, to open a "web admin console" in browser using a port forward. could be an idea here too

**Comment by gtonic** [comment]
> * since the documentation is that great (haha), the idea of the console endpoint was, to open a "web admin console" in browser using a port forward. could be an idea here too

either need to dig around in the codebase or any hint from your side would be appreciated

File: pkg/catalog/keycloak/keycloak.go [hunk 1]
```diff
@@ -67,9 +71,12 @@ func (m *Manager) Info(instance engine.Container) (map[string]string, error) {
 	user := instance.Env["KEYCLOAK_ADMIN"]
 	password := instance.Env["KEYCLOAK_ADMIN_PASSWORD"]
 
+	uri := fmt.Sprintf("localhost:%d/admin", DefaultPort)
```
Comment (gtonic) [inline pkg/catalog/keycloak/keycloak.go]: hm, just spotted, multiple instances (why would one have that?) are not ideally supported with static ports. any basic principles how to deal with that in general?

Comment (adrianliechti) [reply pkg/catalog/keycloak/keycloak.go]: you might find this example useful:
https://github.com/adrianliechti/devkit/blob/main/pkg/catalog/postgres/postgres.go#L72

=> finding the current mapping (local port / static container port) from the runtime-context

Comment (adrianliechti) [reply pkg/catalog/keycloak/keycloak.go]: the console command is a nice hack too:

to expose-non exposed ports (not specified with -p), it starts a helper container with socat and a port-forward, which forwards traffic to the container (withing the vm on your mac/win)

https://github.com/adrianliechti/devkit/blob/main/app/catalog/catalog_cmd.go#L296

**Pull request merged**